packages/dotenv-webpack-plugin/src/index.js

```diff
@@ -0,0 +1,148 @@
+const dotenv = require("dotenv");
+const dotenvExpand = require("dotenv-expand");
+const { DefinePlugin } = require("webpack");
+const { validate } = require("schema-utils");
+const schema = require("./options.json");
+
+/** @typedef {import("./types").Config} Config */
+/** @typedef {import("schema-utils/declarations/validate").Schema} Schema */
+/** @typedef {import("webpack").Compiler} Compiler */
+
+class DotenvWebpackPlugin {
+  /**
+   * Dotenv Webpack Plugin
+   * @param {Config} options - Configuration options
+   */
+  constructor(options = {}) {
+    validate(/** @type {Schema} */ (schema), options || {}, {
+      name: "DotenvWebpackPlugin",
+      baseDataPath: "options",
+    });
+
+    const currentDirectory = process.cwd();
+
+    this.defaultFileList = [
+      `${currentDirectory}/.env.example`, // loaded in all cases
+      `${currentDirectory}/.env`, // loaded in all cases
+      `${currentDirectory}/.env.[mode]`, // only loaded in specified mode
+      `${currentDirectory}/.env.local`, // loaded in all cases, ignored by git
+      `${currentDirectory}/.env.[mode].local`, // only loaded in specified mode, ignored by git
+    ];
+
+    const {
+      // priority is in ascending order
+      // files at the end of the array have higher priority
+      envFiles = this.defaultFileList,
+      prefixes = ["process.env.", "import.meta.env."],
+      envVarPrefix = "PUBLIC_",
```

Comment:
Let's allow it to be `string | string[]`, some env variables can be from other environments, for example SENTRY_ (error reporting) requre prefix `SENTRY_`, and it can be used in client, so you can have `["PUBLIC_", "SENTRY_"]`